matching_test.go

```diff
@@ -0,0 +1,29 @@
+package backoff
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+)
+
+var (
+	customError    = fmt.Errorf("custom error")
+	differentError = fmt.Errorf("different error")
+
+	tests = map[error]error{
+		// sanity check
+		customError:    customError,
+		differentError: differentError,
+		// actual test
+		Permanent(customError):    customError,
+		Permanent(differentError): differentError,
+	}
+)
+
+func Test_ErrorMatching(t *testing.T) {
+	for err, expectedType := range tests {
+		if !errors.Is(err, expectedType) {
```

Comment:
Nice, what about adding new unittest for `errors.As` too?

Reply:
Done.